lib/sequence.js

```diff
@@ -167,6 +167,40 @@ Sequence.prototype._setHooks = function() {
             done();
         }.bind(doc));
     });
+
+    this._schema.pre('findOneAndUpdate', function(next) {
+        // jscs:disable
+        var _thisquery = this,
+            referenceValue;
+
+        // jscs:enable
+
+        this._update['$setOnInsert'] = this._update['$setOnInsert'] || {};
+
+        referenceValue = _this._getCounterReferenceField(this._update['$setOnInsert']);
+        _this._setNextCounter(_this._options.reference_fields, referenceValue, function(err, seq) {
```

Comment:
Either way we'd have to make another query to the db, according to this mongoose issue: https://github.com/Automattic/mongoose/issues/2118

So, I went with make a findOne query in the pre hooks, seems to work fine.